cloud/scope/vpc_cluster.go

```diff
@@ -1646,3 +1774,366 @@ func (s *VPCClusterScope) createSecurityGroupRuleRemote(remote infrav1beta2.VPCS
+	// TODO(cjschaef): Determine if we want to use default LB configuration or require at least one is defined in Cluster spec.
+	// TODO(cjschaef): Remove in favor of webhook validation.
+	if len(s.IBMVPCCluster.Spec.Network.LoadBalancers) == 0 {
+		// We currently don't support any default LB configuration, they must be specified within the Cluster spec.
+		return false, fmt.Errorf("error no load balancers specified for cluster")
+	}
+
+	// Attempt to reconcile each Load Balancer before requeing, if necessary.
+	requeue := false
+	for _, loadBalancer := range s.IBMVPCCluster.Spec.Network.LoadBalancers {
+		// Attempt to retrieve the Load Balancer by Name or ID.
+		lbStatus, err := s.getLoadBalancer(loadBalancer)
+		if err != nil {
+			return false, fmt.Errorf("error retrieving load balancer: %w", err)
+		}
+
+		// If the Load Balancer was found, update Status and move on.
+		if lbStatus != nil {
+			s.setLoadBalancerStatus(lbStatus)
+			// If the Load Balancer status isn't ready, flag for requeue and continue to next Load Balancer.
+			if isReady := s.isLoadBalancerReady(lbStatus.State); !isReady {
+				requeue = true
+			}
+			continue
+		}
+
+		// Otherwise, create the Load Balancer.
+		err = s.createLoadBalancer(loadBalancer)
+		if err != nil {
+			return false, err
+		}
+		// Assume a new Load Balancer will not be ready immediately, due to the complexity and time it takes.
+		requeue = true
+	}
+	return requeue, nil
+}
+
+// isLoadBalancerReady checks the state of a Load Balancer.
+// If state is active, true is returned, in all other cases, it returns false.
+// NOTE(cjschaef): May wish to extend this function to check all Load Balancer details (pools, listeners, etc.) as part of a Load Balancer being ready.
+func (s *VPCClusterScope) isLoadBalancerReady(status infrav1beta2.VPCLoadBalancerState) bool {
+	switch status {
+	case infrav1beta2.VPCLoadBalancerStateActive:
+		s.V(3).Info("load balancer is in active state")
+		return true
+	case infrav1beta2.VPCLoadBalancerStateCreatePending:
+		s.V(3).Info("load balancer is in create pending state")
+	default:
+		s.V(3).Info("load balancer is in unexpected state", "loadBalancerStatus", status)
+	}
+	return false
+}
+
+// getLoadBalancer attempts to retrieve the Load Balancer, otherwise returns nil if it doesn't exist.
+func (s *VPCClusterScope) getLoadBalancer(lb infrav1beta2.VPCLoadBalancerSpec) (*infrav1beta2.VPCLoadBalancerStatus, error) {
+	var loadBalancer *vpcv1.LoadBalancer
+	var err error
+	if lb.ID != nil {
+		var detailedResponse *core.DetailedResponse
+		getLBOptions := &vpcv1.GetLoadBalancerOptions{
+			ID: lb.ID,
+		}
+		loadBalancer, detailedResponse, err = s.VPCClient.GetLoadBalancer(getLBOptions)
+		if detailedResponse != nil && detailedResponse.StatusCode == http.StatusNotFound {
+			return nil, nil
+		}
+	} else {
+		name := lb.Name
+		if name == "" {
+			name = *s.GetServiceName(infrav1beta2.ResourceTypeLoadBalancer)
```

Comment:
In case of multiple loadbalancers configured without name in spec, This may not work as expected as we end up getting same name for all.

Reply:
That is why we do not create an LB with the default Name (let API generate one). But that will generate its own problems.

I will need to rethink how the default service name is being used, as I don't think it makes sense to use it as requested (perhaps only in the rare case that one LB was defined in Spec). Or perhaps I can try refactoring to append `public`/`private` and hope that doesn't cause overlap (although I think that case is just as susceptible to this same situation).
Perhaps I need to add webhook validation to limit the number of LB's defined in Spec to only be 2, in additional to the refactoring. But that can be very limiting to the design and logic as well.

Reply:
In Power VS we do something like this , append the index https://github.com/Karthik-K-N/cluster-api-provider-ibmcloud/blob/987191d98e364d4bbfc39423d433b791dbb6b102/cloud/scope/powervs_cluster.go#L1963-L1965